### DISCUSSION TITLE
How to move a node programmatically

### DISCUSSION THREAD

**Discussion by uto-j** [retejs/rete#591] [Q&A] [answered]
Is there a way to move a node programmatically?
I use react-render-plugin.
I tried below:

>         const nodes = editorRef.current?.nodes?.filter((node) => searchId === node.id);
>         nodes?.forEach((node) => {
>           node.position = newPosition;
>           node.update();
>         });
> 

Then only connections moved.

I can create a new node specified position, but can not move it later.

**Comment by Ni55aN** [comment, marked as answer]
```js
editor.view.nodes.get(node).translate(...newPosition)
```

**Reply by uto-j** [reply]
It's working as expected.
Thank you very much for your help.
I’m really indebted to you for your great work.

**Reply by andre-l-lourenco** [reply]
This may be unrelated to this issue, but my editor object doesn't have a **view** member, am I using a different type?
I've declared it like : `const editor = new NodeEditor<Schemes>();`

**Reply by Ni55aN** [reply]
@andre-l-lourenco the previous answer is for v1. In v2, https://retejs.org/docs/migration#translate-nodes

**Reply by andre-l-lourenco** [reply]
Thanks @Ni55aN
